flake.nix

```diff
@@ -458,61 +478,85 @@
+                  }).result;
+                config-file = writeTextFile {
+                  name = "kusama-local-picasso-dev-karura-dev.json";
+                  text = "${builtins.toJSON config}";
+                };
+              in
+              writeShellApplication {
+                name = "kusama-picasso-karura";
+                text = ''
+                  cat ${config-file}
+                  ${packages.polkadot-launch}/bin/polkadot-launch ${config-file} --verbose
+                '';
               };
-            in writeShellApplication {
-              name = "kusama-picasso-karura";
-              text = ''
-                cat ${config-file}
-                ${packages.polkadot-launch}/bin/polkadot-launch ${config-file} --verbose
-              '';
-            };
 
             default = packages.composable-node;
           };
 
           devShells = rec {
-            developers = mkShell (common-attrs // {
-              buildInputs = with packages; [
-                bacon
-                google-cloud-sdk
-                grub2
-                jq
-                lldb
-                llvmPackages_latest.bintools
-                llvmPackages_latest.lld
-                llvmPackages_latest.llvm
-                mdbook
-                nix-tree
-                nixpkgs-fmt
-                openssl
-                openssl.dev
-                pkg-config
-                qemu
-                rnix-lsp
-                rust-nightly
-                taplo
-                wasm-optimizer
-                xorriso
-                zlib.out
-              ];
-              NIX_PATH = "nixpkgs=${pkgs.path}";
+
+            junod = pkgs.callPackage ./.nix/junod.nix { };
+
+
+            developers = developers-minimal.overrideAttrs (base: {
+              buildInputs = with packages;
+                base.buildInputs ++ [
+                  bacon
+                  google-cloud-sdk
+                  grub2
+                  jq
+                  lldb
+                  llvmPackages_latest.bintools
+                  llvmPackages_latest.lld
+                  llvmPackages_latest.llvm
+                  mdbook
+                  nix-tree
+                  nixpkgs-fmt
+                  openssl
+                  openssl.dev
+                  pkg-config
+                  qemu
+                  rnix-lsp
+                  rust-nightly
+                  taplo
+                  wasm-optimizer
+                  xorriso
+                  zlib.out
+                ];
             });
 
             developers-minimal = mkShell (common-attrs // {
               buildInputs = with packages; [ rust-nightly ];
               NIX_PATH = "nixpkgs=${pkgs.path}";
             });
 
+            developers-xcvm = developers-minimal.overrideAttrs (base: {
+              buildInputs = with packages;
+                base.buildInputs ++
+                [
+                  junod
+                  devnet-dali
```

Comment:
I'm not sure if we need to include junod and devnet-dali in the developer shells. I think these will only be used by other `nix run ...` commands

Reply:
i will use junod for sure. 

why you think that only nix run will use it?